### PR TITLE
chore: Remove IE support

### LINKED ISSUE #654: Get rid of custom IE implementation
Get rid of custom IE implementation (gather a list of these, i.e. Inputs)

### PR DESCRIPTION
**Pull request by adamhock**
- Removed `iui-progress-indicator-radial-ie`
- Removed use of `-ms-grid` in `input-container`
- Removed fallback support for `gap`, `-webkit-line-clamp`, and `appearance`
- Removed `focus-visible` mixin

Closes #654

### CODE REVIEW

**Comment by veekeys** [comment]
Hope after resolving conflicts, it will not show 140 files as changed..

File: src/style/mixins.scss [hunk 1]
```diff
@@ -87,15 +75,11 @@
...
-  white-space: nowrap;
   text-overflow: ellipsis;
-
-  @supports (-webkit-line-clamp: 1) {
-    white-space: unset;
-    display: -webkit-box; /* stylelint-disable-line */
-    -webkit-line-clamp: $lines;
-    -webkit-box-orient: vertical; /* stylelint-disable-line */
-  }
+  white-space: unset;
```
Comment (veekeys) [inline src/style/mixins.scss:79]: should not need to unset anymore as you removed `nowrap`

**Review by veekeys — COMMENTED** [review]
Also update `browserslistrc`

File: src/checkbox/checkbox.scss [hunk 2]
```diff
@@ -19,14 +19,9 @@
 
     gap: $iui-s;
 
-    // #region flex gap fallback
     > * + * {
-      margin-left: $iui-s;
-      @supports (gap: $iui-s) {
-        margin-left: 0;
-      }
+      margin-left: 0;
```
Comment (bentleyvk) [inline src/checkbox/checkbox.scss:23]: So is it needed?

File: src/color-picker/color-picker.scss [hunk 3]
```diff
@@ -26,15 +26,10 @@ $iui-hover-box-shadow: 0 0 1px $iui-xxs + 1 rgba(var(--iui-color-foreground-body
...
-    row-gap: $iui-baseline;
-  }
+  row-gap: $iui-baseline;
 
   > div:not(:last-child) {
-    margin-bottom: $iui-baseline;
-    @supports (row-gap: $iui-baseline) {
-      margin-bottom: 0;
-    }
+    margin-bottom: 0;
```
Comment (bentleyvk) [inline src/color-picker/color-picker.scss:32]: Again, if we use gap, is this needed?

File: src/color-picker/color-picker.scss [hunk 4]
```diff
@@ -45,16 +40,12 @@ $iui-hover-box-shadow: 0 0 1px $iui-xxs + 1 rgba(var(--iui-color-foreground-body
   width: $iui-l;
   cursor: pointer;
   border-radius: $iui-color-swatch-border-radius;
-  margin-bottom: $iui-xs;
-  margin-right: $iui-xs;
+  margin-bottom: 0;
```
Comment (bentleyvk) [inline src/color-picker/color-picker.scss:43]: Remove?

File: src/color-picker/color-picker.scss [hunk 5]
```diff
@@ -131,10 +120,7 @@ $iui-hover-box-shadow: 0 0 1px $iui-xxs + 1 rgba(var(--iui-color-foreground-body
     }
 
     > *:not(:last-child) {
-      margin-right: $iui-xxs;
-      @supports (gap: $iui-xxs) {
-        margin-right: 0;
-      }
+      margin-right: 0;
```
Comment (bentleyvk) [inline src/color-picker/color-picker.scss:123]: Not needed?

File: src/dialog/dialog.scss [hunk 6]
```diff
@@ -192,22 +192,12 @@
...
   align-items: center;
   justify-content: flex-end;
-
+  column-gap: $iui-s;
   @at-root .iui-dialog-draggable & {
     padding: 0 $iui-m $iui-baseline $iui-m;
   }
 
-  // #region IE support
   > .iui-button:not(:last-child) {
```
Comment (bentleyvk) [inline src/dialog/dialog.scss:200]: Shouldn't be needed when gap is used

File: src/input-container/input-container.scss [hunk 7]
```diff
@@ -131,11 +79,7 @@
       > .iui-checkbox-wrapper,
       > .iui-radio-wrapper,
       > .iui-toggle-switch-wrapper {
-        margin-right: $iui-m;
-
-        @supports (gap: $iui-m) {
-          margin-right: 0;
-        }
+        margin-right: 0;
```
Comment (bentleyvk) [inline src/input-container/input-container.scss:82]: Not needed

File: src/input-container/input-container.scss [hunk 8]
```diff
@@ -131,11 +79,7 @@
...
       }
 
       &:not(:last-child) {
```
Comment (bentleyvk) [inline src/input-container/input-container.scss:85]: I think this should also be removed

File: src/toggle-switch/toggle-switch.scss [hunk 9]
```diff
@@ -18,10 +18,7 @@
   }
 
   > * + * {
-    margin-left: $iui-s;
-    @supports (gap: $iui-s) {
-      margin-left: 0;
-    }
+    margin-left: 0;
```
Comment (bentleyvk) [inline src/toggle-switch/toggle-switch.scss:21]: Remove?

**Comment by bentleyvk** [comment]
> I think we will need to keep the `focus-visible` mixin because it is a _very_ new addition to Safari.
> 
> We might even want to keep some of the flex `gap` fallbacks because it was only added last year, although `@supports (gap: 2px)` is not entirely reliable because it will return true when grid gap is supported and flex gap is not supported. One workaround here could be to replace flex with grid in these places.

Is one year not enough with `gap` support is not enough?

**Comment by mayank99** [comment]
> Is one year with `gap` support is not enough?

It's not bad but still feels kinda new. If we used gap with `grid` instead of `flex`, we could get it up to 93%.

but ultimately it's just whitespace so I'm fine either way.

**Review by elephantcatdog — COMMENTED** [review]
The other comments have covered everything I noticed.

File: src/input-container/input-container.scss (hunk 8 above)
Comment (adamhock) [reply src/input-container/input-container.scss:79]: I tried removing this one and it lead to some visual changes. I'll look into it a little further

File: src/input-container/input-container.scss (hunk 7 above)
Comment (adamhock) [reply src/input-container/input-container.scss]: fixed

File: src/dialog/dialog.scss (hunk 6 above)
Comment (adamhock) [reply src/dialog/dialog.scss]: fixed

File: src/color-picker/color-picker.scss (hunk 5 above)
Comment (adamhock) [reply src/color-picker/color-picker.scss]: fixed

File: src/color-picker/color-picker.scss (hunk 4 above)
Comment (adamhock) [reply src/color-picker/color-picker.scss]: fixed

File: src/color-picker/color-picker.scss (hunk 3 above)
Comment (adamhock) [reply src/color-picker/color-picker.scss]: fixed

File: src/checkbox/checkbox.scss (hunk 2 above)
Comment (adamhock) [reply src/checkbox/checkbox.scss]: fixed

File: src/style/mixins.scss (hunk 1 above)
Comment (adamhock) [reply src/style/mixins.scss]: fixed

**Comment by adamhock** [comment]
> > Is one year with `gap` support is not enough?
> 
> It's not bad but still feels kinda new. If we used gap with `grid` instead of `flex`, we could get it up to 93%.
> 
> but ultimately it's just whitespace so I'm fine either way.

I have a local stash with changes that change the majority of the flex gap changes to grid gap. I'm still trying to fix one or two more cases in color picker.

File: .browserslistrc [hunk 10]
```diff
@@ -1,12 +1,14 @@
 # This file is used by autoprefixer and stylelint to adjust CSS output to support the specified browsers below.
+
```
Comment (bentleyvk) [inline .browserslistrc:2]: Just remove those new lines for less changes.

File: src/radio-tile/radio-tile.scss [hunk 11]
```diff
@@ -68,6 +68,7 @@
   appearance: none;
   opacity: 0;
   position: absolute;
+
```
Comment (bentleyvk) [inline src/radio-tile/radio-tile.scss:71]: Remove this new line so it won't be shown as a change

File: src/style/mixins.scss [hunk 12]
```diff
@@ -19,7 +19,6 @@
     text-decoration: none;
   }
 }
-
```
Comment (bentleyvk) [inline src/style/mixins.scss:22]: Bring back the ne lines

File: src/toggle-switch/toggle-switch.scss (hunk 9 above)
Comment (bentleyvk) [reply src/toggle-switch/toggle-switch.scss:21]: Is it?

File: src/style/mixins.scss (hunk 12 above)
Comment (adamhock) [reply src/style/mixins.scss]: fixed

File: src/toggle-switch/toggle-switch.scss (hunk 9 above)
Comment (adamhock) [reply src/toggle-switch/toggle-switch.scss]: Now it is

File: scripts/generateCss.js [hunk 13]
```diff
@@ -18,10 +18,10 @@ const compileScss = async (path, outFile) => {
   return new Promise(async (resolve, reject) => {
     try {
       const cssResult = await sass.compileAsync(path);
-      const processedCssResult = await postcss([
-        require('autoprefixer')({ grid: 'autoplace' }),
-        require('postcss-discard-comments'),
-      ]).process(cssResult.css, { from: undefined }); // `from` is required
+      const processedCssResult = await postcss([require('autoprefixer'), require('postcss-discard-comments')]).process(
```
[the remark this answers is not in the document]
Comment (adamhock) [reply scripts/generateCss.js]: Fixed

**Review by veekeys — APPROVED** [review]

**Comment by bentleyvk** [comment]
So are we going to replace all `:not(:last-child)` that I mentioned in previous comment and that were done because of IE or not?

**Comment by adamhock** [comment]
> * color-picker
>   image

I kept the use of `margin-bottom` for `iui-color-field` because not every child of `iui-color-selection-wrapper` is going to be `iui-color-field`, so we only want that margin if the color field is the last child

> * information-panel
>   image

I'm not sure if there's a way to use `gap` for this because this margin is applied to every other child so that the labels and inputs are grouped together

> * column-filter
>   image
> * pill tabs
>   image
> * tags
>   image
> * tile
>   image

These instances have been changed to incorporate `gap`

File: src/input-container/input-container.scss (hunk 8 above)
Comment (adamhock) [reply src/input-container/input-container.scss]: This use of `margin-right` is to separate `iui-input-group` from `iui-end-icon`, so it must be kept.

File: packages/itwinui-css/src/tag/tag.scss [hunk 14]
```diff
@@ -59,6 +59,7 @@
 
 @mixin iui-tag-container {
   color: var(--iui-text-color-muted);
+  column-gap: $iui-xs;
```
Comment (bentleyvk) [inline packages/itwinui-css/src/tag/tag.scss]: Don't you need to remove?
```sass
> *:not(last-child) {
  margin-right: $iui-xs
}
```

Comment (adamhock) [reply packages/itwinui-css/src/tag/tag.scss]: Actually, because `iui-tag-container` does not have `display: flex`, I cannot use `gap` here

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**